### PR TITLE
feat: add a default deployable region

### PR DESCRIPTION
**Pull request by gnhenderson**
### Changes
- Add new `SLINGSHOT_ENGINE` config parameter `default_endpoint` to be used when no explicit `regions` have been defined in a slingshot yaml.
  - When no explicit `regions:` are defined in a slingshot yaml default API behaviour will be to trigger a deployment to the engine endpoint defined in `default_endpoint`.
  - If no `default_endpoint` config value is defined and no explicit `regions:` are declared in a slingshot yaml an exception is raised and the deployment fails.
- Slingshot yaml now accepts the following styles of legal values for `regions:`
  ```
  ...
  regions:
    eu-west-1:
      desired_count: 2
      ... 
  ...
  ```
  or
  ```
  ...
  regions:
    eu-west-1: {}
    eu-central-1: {}
  ...
  ```
  or
  ```
  ...
  regions:
    - eu-west-1
      eu-central-1
  ...
  ```
  or
  ```
  ...
  regions: {}
  ...
  ```
  or `regions:` can be completely left out of the slingshot yaml file (and the `default_endpoint` config value will be used).

### CODE REVIEW

File: config/app/default.py [hunk 1]
```diff
@@ -21,7 +21,11 @@
         # 'localhost': http://localhost:5001,
```
Comment (catscanner) [inline config/app/default.py]: Just a note, but actually that needs to be 'localhost': 'http://localhost:5001', for actual accuracy as it needs to be a quoted string.

File: slingshotapi/deploy.py [hunk 2]
```diff
@@ -25,9 +28,10 @@ class DatabaseConnectionError(Exception):
 
     @classmethod
     def deploy(cls, deployment_data):
-        regions = deployment_data.get("regions", {})
-        overrides = deployment_data.get("override", {})
-        project = deployment_data.get("project", "unknown")
+        default_region = flask.current_app.config.get('SLINGSHOT_ENGINE', {}).get('default_endpoint', None)
+        regions = deployment_data.get('regions', default_region) or default_region
```
Comment (catscanner) [inline slingshotapi/deploy.py]: What condition leads the 'or' to be reached?

Comment (gnhenderson) [inline slingshotapi/deploy.py]: If `regions` is an empty dictionary we want to assign the `default_region` value. To do that we first need to evaluate whether `deployment_data.get('regions', default_region)` returns empty or not. If `regions` exists and is an empty dict then the `.get()` won't automagically assign the `default_region` value. The `or` forces the empty dictionary replacement.

Comment (catscanner) [inline slingshotapi/deploy.py]: thanks.

**Comment by catscanner** [comment]
seems ok

**Pull request merged**